### PR TITLE
Fixes initial damages not being applied on flamer_fire instanciation

### PR DESCRIPTION
**Pull request by Menshin**
## About The Pull Request

Quickfix for initial damages on flame creation being overlooked in the fire range refactor.
Some light code cleaning, but the whole flame code is a mess and should be added on the refactor list, if not already.

## Changelog
:cl:
fix: Flame now applies initial damage correctly

/:cl:

### CODE REVIEW

File: code/modules/projectiles/updated_projectiles/guns/flamer.dm [hunk 1]
```diff
@@ -457,52 +462,39 @@
...
 	return ..()
 
+/obj/flamer_fire/proc/apply_initial_damage(fire_stacks, fire_damage)
+	var/turf/T = get_turf(src)
+
+	for(var/mob/living/C in T)
+		if(C.fire_immune)
+			continue
+		else
+			C.adjust_fire_stacks(fire_stacks)
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/guns/flamer.dm]: screammmmmmmmmmmmmmmmm use the flamer_fire_act() etc mob procs.

File: code/modules/projectiles/updated_projectiles/guns/flamer.dm [hunk 2]
```diff
@@ -457,52 +462,38 @@
...
 /obj/flamer_fire/Destroy()
 	SetLuminosity(0)
 	STOP_PROCESSING(SSobj, src)
 	return ..()
 
+/obj/flamer_fire/proc/apply_initial_damage(fire_stacks, fire_damage)
+	var/turf/T = get_turf(src)
+
+	for(var/mob/living/C in T)
+		if(C.fire_immune)
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/guns/flamer.dm]: should be handled by flamer_fire_act

Comment (Menshin) [reply code/modules/projectiles/updated_projectiles/guns/flamer.dm]: Except it doesn't, want me to add it ?

Comment (spookydonut) [reply code/modules/projectiles/updated_projectiles/guns/flamer.dm]: if it isnt it should be handled by it, xenos do fire immune differently from memory so its important that behaviour is done via override.

even if you make a new proc called like flamer_fire_direct_act or something

**Comment by Menshin** [comment]
Indeed, the flame code deserves a real refactor instead of a quick fix. I'll close this PR until i have some time to do it properly (still on area/lighting/machinery refactor). Of course, if anyone wants to take the task, be my guest.

**Pull request closed without merging**